src/pm/xbps.rs

```diff
@@ -0,0 +1,256 @@
+        if kws.is_empty() {
+            return self
+                .run(Cmd::new(&["xbps-query", "-l"]).kws(kws).flags(flags))
+                .await;
+        }
+
+        let lines: Vec<_> = stream::iter(kws)
+            .map(Ok)
+            .and_then(|&pkg| async {
+                let cmd = Cmd::new(&["xbps-query", "--property", "pkgver", pkg]).flags(flags);
+                self.check_output(cmd, PmMode::Mute, &Strategy::default())
+                    .await
+                    .map_err(|err| (pkg.to_string(), err))
+            })
+            .collect()
+            .await;
+
+        let mut stdout = tokio::io::stdout();
+        for line in lines {
+            match line {
+                Ok(line) => stdout.write_all(&line).await?,
+                Err((
+                    pkg,
+                    err @ Error::CmdStatusCodeError {
+                        code: PKG_NOT_FOUND_CODE,
+                        ..
+                    },
+                )) => {
+                    let msg = format!("package '{}' was not found", pkg);
+                    print_err(err, &msg);
+                }
+                Err((_, other)) => return Err(other),
+            };
+        }
+
+        Ok(())
+    }
+
+    /// Qe lists packages installed explicitly (not as dependencies).
+    async fn qe(&self, kws: &[&str], flags: &[&str]) -> Result<()> {
+        if kws.is_empty() {
+            return self
+                .run(Cmd::new(&["xbps-query", "-m"]).kws(kws).flags(flags))
+                .await;
+        }
+
+        let lines: Vec<_> = stream::iter(kws)
+            .filter(|&&pkg| async {
+                let check_cmd =
+                    Cmd::new(&["xbps-query", "--property", "automatic-install", pkg]).flags(flags);
+                let result = self
+                    .check_output(check_cmd, PmMode::Mute, &Strategy::default())
+                    .await;
+
+                // if a package is manually installed then the automatic-install field is empty
+                match result {
+                    Ok(auto_status) => auto_status.is_empty(),
+                    _ => true,
+                }
+            })
+            .map(Ok)
+            .and_then(|&pkg| async {
+                let cmd = Cmd::new(&["xbps-query", "--property", "pkgver", pkg]).flags(flags);
+                self.check_output(cmd, PmMode::Mute, &Strategy::default())
+                    .await
+                    .map_err(|err| (pkg.to_string(), err))
+            })
+            .collect()
+            .await;
+
+        let mut stdout = tokio::io::stdout();
+        for line in lines {
+            match line {
+                Ok(line) => stdout.write_all(&line).await?,
+                Err((
+                    pkg,
+                    err @ Error::CmdStatusCodeError {
+                        code: PKG_NOT_FOUND_CODE,
+                        ..
+                    },
+                )) => {
```

Comment:
One final tiny detail here: in order to be coherent with the previous https://github.com/rami3l/pacaptr/issues/25 fix, it's required to finally return:
- If all subprocesses returned `0`, then return `Ok(())`.
- If all subprocesses are of exit code `2` (some can also be `0`), then one `CmdStatusCodeError { code: PKG_NOT_FOUND_CODE, .. }` should be returned. You can take an error message like "some package(s) are not found".
- Otherwise, return the first error as you normally would.

Same thing for `-Q`, I guess.

Reply:
One more case: if there are only `Ok`s and `PKG_NOT_FOUND`s, the function should return a `CmdStatusCodeError`?

Reply:
@rami3l should now work as specified